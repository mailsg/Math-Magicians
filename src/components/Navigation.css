.navigation {
  display: flex;
  padding: 20px;
  align-items: center;
  justify-content: space-between;
  background-color: chocolate;
}

h1 {
  font-size: 54px;
  margin: 20px;
}

.item {
  display: flex;
  font-size: 24px;
  justify-content: flex-end;
  align-items: flex-end;
  gap: 20px;
  list-style: none;
  margin-right: 1rem;
}

.links {
  text-decoration: none;
  color: black;
}

.links:hover {
  color: blue;
}
