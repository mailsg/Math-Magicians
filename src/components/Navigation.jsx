import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navigation.css';

const Navbar = [
  { to: '/', text: 'Home  |' },
  { to: '/calculator', text: 'Calculator  |' },
  { to: '/hittheapi', text: 'Quote  |' },
];

const Nav = () => (
  <nav className="navigation">
    <h1>Math Magicians</h1>
    <ul className="item">
      {Navbar.map((link) => (
        <li key={link.to}>
          <NavLink to={link.to} className="links">
            {link.text}
          </NavLink>
        </li>
      ))}
    </ul>
  </nav>
);

export default Nav;
