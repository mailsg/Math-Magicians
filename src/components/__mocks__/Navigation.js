import React from 'react';
import { BrowserRouter, NavLink } from 'react-router-dom';
import '../Navigation.css';

const Navbar = [
    { to: '/', text: 'Home  |' },
    { to: '/calculator', text: 'Calculator  |' },
    { to: '/hittheapi', text: 'Quote  |' },    
];

const Nav = () => (
    <BrowserRouter>
        <nav className="navigation">
            <h1>Math Magicians</h1>
            <ul className="item">
                {Navbar.map((link) => (
                    <li key={link.to}>
                        <NavLink to={link.to} className="links">
                            {link.text}
                        </NavLink>
                    </li>
                ))}
            </ul>
        </nav>
    </BrowserRouter>
);

export default Nav;
