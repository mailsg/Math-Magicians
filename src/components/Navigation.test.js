import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from './__mocks__/Navigation';

it('Navigation renders correctly', () => {
    const tree = renderer
        .create(<Navigation />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});