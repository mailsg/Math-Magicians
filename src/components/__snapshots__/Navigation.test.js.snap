// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<nav
  className="navigation"
>
  <h1>
    Math Magicians
  </h1>
  <ul
    className="item"
  >
    <li>
      <a
        aria-current="page"
        className="links active"
        href="/"
        onClick={[Function]}
      >
        Home  |
      </a>
    </li>
    <li>
      <a
        className="links"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator  |
      </a>
    </li>
    <li>
      <a
        className="links"
        href="/hittheapi"
        onClick={[Function]}
      >
        Quote  |
      </a>
    </li>
  </ul>
</nav>
`;
